packages/cojson/src/PeerState.ts

```diff
@@ -150,4 +153,35 @@ export class PeerState {
     this.peer.outgoing.close();
     this.closed = true;
   }
+
+  async processIncomingMessages(callback: (msg: SyncMessage) => Promise<void>) {
```

Comment:
Moved this logic inside PeerState to handle the tracking of incomingMessagesProcessingPromise internally